tsconfig.json

```diff
@@ -1,9 +1,23 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
-    "lib": ["es2015"],
-    "target": "es2015"
+    "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
+    "experimentalDecorators": true,
+    "importHelpers": true,
+    "lib": [
+      "es2018",
+    ],
+    "moduleResolution": "node",
+    "noEmitHelpers": true,
+    "noEmitOnError": true,
+    "noImplicitReturns": true,
+    "target": "es5",
```

Comment:
Is there a reason to change the target from es2015 to es5?

Reply:
oh, good catch. This was not meant to change.

Context: This branch started off as two features, this feature but then also another one which made the tests run on node v8.x - during that development I had an issue where the tests were not running on v6, I changed the value, it didn't make a difference, and I forgot to change it base.

The issue was that I'd switched the entire project over to using ts-jest at some point and it complained about some of the javascript within the project, but only on v6 - so I figured something about targeting or babel was wrong, and I was iterating through that.

I'll go ahead and change it back to es2015.